packages/near-api-js/src/contract.ts

```diff
@@ -90,45 +119,92 @@ export class Contract {
     constructor(account: Account, contractId: string, options: ContractMethods) {
         this.account = account;
         this.contractId = contractId;
-        const { viewMethods = [], changeMethods = [] } = options;
-        viewMethods.forEach((methodName) => {
-            Object.defineProperty(this, methodName, {
+        const { viewMethods = [], changeMethods = [], abi: abiRoot } = options;
```

Comment:
Is there a use case for calling the constructor with `options` that have values in `viewMethods`/`changeMethods` and also methods specified in `abiRoot.body.functions`? I see that if there is any overlap in method names the constructor will break trying to redefine properties.

It seems odd to now have two ways of initializing contract methods. If they are mutually exclusive then maybe it makes more sense to only source methods from `abiRoot.body.functions` when provided. This is basically equivalent to changing the constructor signature to `Contract(Account, string, ContractMethods | AbiRoot)` which feels a bit more deliberate.

What do you think?

Reply:
Yeah, so my intention was to leave an option to provide ABI **and** some custom methods on the off-chance that they are not in the ABI. But I guess this might not make sense until we actually encounter a situation where a user needs this, so I would be totally down to change this to `ContractMethods | AbiRoot`.

Reply:
How likely is it that an ABI wouldn't include all methods? I'm not really familiar with how you would get the ABI in the first place, i.e. if you create them dynamically from a deployed contract or need to generate them statically.

Reply:
You generate them statically from the source code. Assuming there are no bugs in the ABI generation code it should cover all methods exposed by the contract. That being said, it does not stop the user from injecting new methods into the resulting wasm through some postprocessing means.

Reply:
So, overall, I would say it is fairly unlikely to happen (I have never seen this pattern among our users), and I guess if you are really in a pickle, you could just instantiate two separate `Contract`s for now.

Reply:
Ok, I went ahead and prohibited the conflicting usage of `viewMethods`/`changeMethods` with ABI. LMK if you disagree.